### PR TITLE
Added changelog for `0.61.0` and updated release versioning table

### PR DESCRIPTION
**Pull request by kc611**
As titled, this PR adds changelog for `0.61.0` and update the release versioning table therein.

### CODE REVIEW

**Review by gmarkall — DISMISSED** [review]
CUDA-related wording looks good.

File: docs/source/release/0.61.0-notes.rst [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+preferred place to get CUDA support from this release onwards.
+LLVM 15 is now supported as the default LLVM version. 
+
+Please find a summary of all noteworthy items below.
+
+Highlights
+~~~~~~~~~~
+
+Add initial implementation for a new type system.
+=================================================
```
Comment (sklam) [inline docs/source/release/0.61.0-notes.rst:21]: Wrong heading level

Screenshot 2024-11-20 at 9 07 26 AM

File: docs/source/release/0.61.0-notes.rst [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+with an iterative one. This removes a risk of hitting the Python recursion limit.
+
+(`PR-#9718 <https://github.com/numba/numba/pull/9718>`__)
+
+
+NumPy Support
+~~~~~~~~~~~~~
+
+Added support for np.setdiff1d(), np.setxor1d(), and np.in1d() functions, as well as argument assume_unique in np.intersect1d()
+===============================================================================================================================
```
Comment (sklam) [inline docs/source/release/0.61.0-notes.rst:178]: Wrong heading level

File: docs/source/release/0.61.0-notes.rst [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+
+- Retain Numba's old type system as default.
+- Add a config flag `USE_LEGACY_TYPE_SYSTEM` set to `1` (on) by default.
+  Switching it to `0` (off) will activate new type system.
+- Within the new type system, Python and NumPy scalars will be treated
+  and returned separately as different entities through JIT compiled functions.
+
+(`PR-#9662 <https://github.com/numba/numba/pull/9662>`__)
+
+Python 3.13 support
```
Comment (sklam) [inline docs/source/release/0.61.0-notes.rst:36]: Azure error: https://dev.azure.com/numba/numba/_build/results?buildId=18003&view=logs&j=92a5d2dc-b1c1-5fd3-be4c-f2c5b5a37a8d&t=e3dfe365-232a-52a4-3ae3-57b751ae31cd&l=177

File: docs/source/release/0.61.0-notes.rst (hunk 1 above)
Comment (sklam) [reply docs/source/release/0.61.0-notes.rst:21]: Link to page: https://numba--9800.org.readthedocs.build/en/9800/release-notes-overview.html

**Review by sklam — APPROVED** [review]

File: docs/source/release/0.61.0-notes.rst [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+Add initial implementation for a new type system.
+-------------------------------------------------
+
+This adds a new type system that will allow Numba to
+differentiate between Python and NumPy scalars.
+
+This has been achieved as follows:
+
+- Retain Numba's old type system as default.
+- Add a config flag `USE_LEGACY_TYPE_SYSTEM` set to `1` (on) by default.
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:29]:
```suggestion
- Add a config flag ``USE_LEGACY_TYPE_SYSTEM`` set to ``1`` (on) by default.
```

File: docs/source/release/0.61.0-notes.rst [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+  Switching it to `0` (off) will activate new type system.
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:30]:
```suggestion
  Switching it to ``0`` (off) will activate new type system.
```

File: docs/source/release/0.61.0-notes.rst [hunk 6]
```diff
@@ -0,0 +1,457 @@
...
+(`PR-#9741 <https://github.com/numba/numba/pull/9741>`__)
+
+
+New Features
+~~~~~~~~~~~~
+
+Schedule guvectorize'd functions over dask.distributed
+------------------------------------------------------
+
+Functions decorated with `@guvectorize` can now be scheduled over distributed
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:71]:
```suggestion
Functions decorated with ``@guvectorize`` can now be scheduled over distributed
```

File: docs/source/release/0.61.0-notes.rst [hunk 7]
```diff
@@ -0,0 +1,457 @@
...
+
+Move creation of ``ModulePassManager`` object to ``_optimize_final_module``
+function, preventing the usage of the same pass manager object for
+compiling multiple Python functions. This would allow for better control
+while optimizing unrelated modules and possibly under different settings
+(degree of vectorization, optimization level, etc.).
+
+(`PR-#9670 <https://github.com/numba/numba/pull/9670>`__)
+
+Fixed ``typed.List __repr__()`` to display ellipsis appropriately.
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:146]:
```suggestion
Fixed ``typed.List __repr__()`` to display ellipsis appropriately
```

File: docs/source/release/0.61.0-notes.rst [hunk 8]
```diff
@@ -0,0 +1,457 @@
...
+------------------------------------------------------------------
+
+``typed.List __repr__()`` has been fixed to display the list elements
+without appending the ellipsis at the end, up until a maximum of 1000 elements.
+Previously, the list repr would append the ellipsis at the end of the list
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:151]:
```suggestion
Previously, the list ``repr`` would append the ellipsis at the end of the list
```

File: docs/source/release/0.61.0-notes.rst [hunk 9]
```diff
@@ -0,0 +1,457 @@
...
+
+Use of dead branch pruning improved
+-----------------------------------
+
+Dead-branch pruning use is improved to support cases when the predicate
+expression is dependent on a variable that later changes type.
+
+(`PR-#9711 <https://github.com/numba/numba/pull/9711>`__)
+
+``find_topo_sort`` reliablilty improvement
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:164]:
```suggestion
``find_topo_sort`` reliabililty improvement
```

File: docs/source/release/0.61.0-notes.rst [hunk 10]
```diff
@@ -0,0 +1,457 @@
...
+
+``memcpy`` static buffer content into newly allocated buffer
+------------------------------------------------------------
+
+Fix a bug where the static buffer used to store typecode representation is not
+copied to the new allocated buffer as part of a realloc operation.
+
+(`PR-#9119 <https://github.com/numba/numba/pull/9119>`__)
+
+Fix parfor hoisting.
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:205]:
```suggestion
Fix parfor hoisting
```

File: docs/source/release/0.61.0-notes.rst [hunk 11]
```diff
@@ -0,0 +1,457 @@
...
+--------------------
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:206]:
```suggestion
-------------------
```

File: docs/source/release/0.61.0-notes.rst [hunk 12]
```diff
@@ -0,0 +1,457 @@
...
+preferred place to get CUDA support from this release onwards.
+LLVM 15 is now supported as the default LLVM version. 
+
+Please find a summary of all noteworthy items below.
+
+Highlights
+~~~~~~~~~~
+
+Add initial implementation for a new type system.
+-------------------------------------------------
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:21]:
```suggestion
------------------------------------------------
```

File: docs/source/release/0.61.0-notes.rst [hunk 13]
```diff
@@ -0,0 +1,457 @@
...
+Move creation of ``ModulePassManager`` object to ``_optimize_final_module``
+function, preventing the usage of the same pass manager object for
+compiling multiple Python functions. This would allow for better control
+while optimizing unrelated modules and possibly under different settings
+(degree of vectorization, optimization level, etc.).
+
+(`PR-#9670 <https://github.com/numba/numba/pull/9670>`__)
+
+Fixed ``typed.List __repr__()`` to display ellipsis appropriately.
+------------------------------------------------------------------
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:147]:
```suggestion
-----------------------------------------------------------------
```

File: docs/source/release/0.61.0-notes.rst [hunk 14]
```diff
@@ -0,0 +1,457 @@
...
+Support is added for: ``numpy.setdiff1d()``, ``np.setxor1d()``, ``np.in1d()``, and ``np.isin()``; and the argument ``assume_unique`` in ``np.intersect1d()``.
+For ``np.in1d()``, and ``np.isin()``, the keyword ``kind`` is *not* supported, and the behaviour reflects that of NumPy prior to version 1.24.
+This is equivalent to setting ``kind="sort"`` in NumPy 1.24 and later.
+
+(`PR-#9338 <https://github.com/numba/numba/pull/9338>`__)
+
+Support for np.trapezoid
+------------------------
+
+Add support for numpy 2.0 new function ``numpy.trapezoid``.
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:189]:
```suggestion
Add support for NumPy 2.0 new function ``numpy.trapezoid``.
```

File: docs/source/release/0.61.0-notes.rst [hunk 15]
```diff
@@ -0,0 +1,457 @@
...
+Added support for np.setdiff1d(), np.setxor1d(), and np.in1d() functions, as well as argument assume_unique in np.intersect1d()
+-------------------------------------------------------------------------------------------------------------------------------
+
+Support is added for: ``numpy.setdiff1d()``, ``np.setxor1d()``, ``np.in1d()``, and ``np.isin()``; and the argument ``assume_unique`` in ``np.intersect1d()``.
+For ``np.in1d()``, and ``np.isin()``, the keyword ``kind`` is *not* supported, and the behaviour reflects that of NumPy prior to version 1.24.
+This is equivalent to setting ``kind="sort"`` in NumPy 1.24 and later.
+
+(`PR-#9338 <https://github.com/numba/numba/pull/9338>`__)
+
+Support for np.trapezoid
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:186]:
```suggestion
Support for ``np.trapezoid``
```

File: docs/source/release/0.61.0-notes.rst [hunk 16]
```diff
@@ -0,0 +1,457 @@
...
+------------------------
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:187]:
```suggestion
----------------------------
```

File: docs/source/release/0.61.0-notes.rst [hunk 17]
```diff
@@ -0,0 +1,457 @@
...
+-----------------------------------------------------------------------------------------------
+
+In the case of repeated dispatch on non-fingerprintable types, the dispatcher
+now uses memory in proportion to the number of unique types seen opposed to in
+proportion to the number of types in total.
+
+(`PR-#9757 <https://github.com/numba/numba/pull/9757>`__)
+
+Fix miscompile in branch pruning of SSA form IR.
+------------------------------------------------
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:282]:
```suggestion
-----------------------------------------------
```

File: docs/source/release/0.61.0-notes.rst [hunk 18]
```diff
@@ -0,0 +1,457 @@
...
+
+A miscompile occurring when a binop expression with constant arguments is used
+as a predicate in SSA form IR is now fixed.
+
+(`PR-#9758 <https://github.com/numba/numba/pull/9758>`__)
+
+Fix regression in type-inference for star-arg arguments.
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:289]:
```suggestion
Fix regression in type-inference for star-arg arguments
```

File: docs/source/release/0.61.0-notes.rst [hunk 19]
```diff
@@ -0,0 +1,457 @@
...
+--------------------------------------------------------
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:290]:
```suggestion
-------------------------------------------------------
```

File: docs/source/user/installing.rst [hunk 20]
```diff
@@ -266,7 +266,8 @@ information.
 +----------++--------------+---------------------------+---------------------------------------------+------------------------------+-------------------+-----------------------------+
 | Numba     | Release date | Python                    | NumPy                                       | llvmlite                     | LLVM              | TBB                         |
 +===========+==============+===========================+=============================================+==============================+===================+=============================+
-| 0.61.0    | UNRELEASED   | 3.10.x <= version < 3.14  | 1.24 <= version < 1.27 ; version == 2.0 †   | 0.43.x                       | 14.x              | 2021.6 <= version           |
+| 0.61.0    | 2024-11-25   | 3.10.x <= version < 3.14  | 1.24 <= version < 1.27 ;                    | 0.44.x                       | 15.x              | 2021.6 <= version           |
```
Comment (esc) [inline docs/source/user/installing.rst:269]:
```suggestion
| 0.61.0rc1 | 2024-11-25   | 3.10.x <= version < 3.14  | 1.24 <= version < 1.27 ;                    | 0.44.x                       | 15.x              | 2021.6 <= version           |
```

File: docs/source/release/0.61.0-notes.rst [hunk 21]
```diff
@@ -0,0 +1,457 @@
...
+
+This release adds support for NumPy 2.1 (excluding the NEP-050 semantics).
+
+(`PR-#9741 <https://github.com/numba/numba/pull/9741>`__)
+
+
+New Features
+~~~~~~~~~~~~
+
+Schedule guvectorize'd functions over dask.distributed
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:68]:
```suggestion
Schedule guvectorize'd functions over ``dask.distributed``
```

File: docs/source/release/0.61.0-notes.rst [hunk 22]
```diff
@@ -0,0 +1,457 @@
...
+------------------------------------------------------
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:69]:
```suggestion
----------------------------------------------------------
```

File: docs/source/release/0.61.0-notes.rst [hunk 23]
```diff
@@ -0,0 +1,457 @@
...
+(typically through generated code) by replacing the recursive post order traversal
+with an iterative one. This removes a risk of hitting the Python recursion limit.
+
+(`PR-#9718 <https://github.com/numba/numba/pull/9718>`__)
+
+
+NumPy Support
+~~~~~~~~~~~~~
+
+Added support for np.setdiff1d(), np.setxor1d(), and np.in1d() functions, as well as argument assume_unique in np.intersect1d()
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:177]:
```suggestion
Added support for ``np.setdiff1d()``, ``np.setxor1d()``, and ``np.in1d()`` functions, as well as argument ``assume_unique`` in ``np.intersect1d()``
```

File: docs/source/release/0.61.0-notes.rst [hunk 24]
```diff
@@ -0,0 +1,457 @@
...
+-------------------------------------------------------------------------------------------------------------------------------
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:178]:
```suggestion
-------------------------------------------------------------------------------------------------------------------------------------------------
```

Comment (esc) [reply docs/source/release/0.61.0-notes.rst:178]: Please check the number of hyphens here after applying the suggestion, thanks. It should be 20.

File: docs/source/release/0.61.0-notes.rst [hunk 25]
```diff
@@ -0,0 +1,457 @@
+
+Version 0.61.0 (25 November 2024)
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:2]:
```suggestion
Version 0.61.0rc1 (25 November 2024)
```

File: docs/source/release/0.61.0-notes.rst [hunk 26]
```diff
@@ -0,0 +1,457 @@
...
+=================================
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:3]:
```suggestion
====================================
```

File: docs/source/release/0.61.0-notes.rst [hunk 27]
```diff
@@ -0,0 +1,457 @@
...
+with an iterative one. This removes a risk of hitting the Python recursion limit.
+
+(`PR-#9718 <https://github.com/numba/numba/pull/9718>`__)
+
+
+NumPy Support
+~~~~~~~~~~~~~
+
+Added support for ``np.setdiff1d()``, ``np.setxor1d()``, and ``np.in1d()`` functions, as well as argument ``assume_unique`` in ``np.intersect1d()``
+-------------------------------------------------------------------------------------------------------------------------------------------------
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:178]:
```suggestion
---------------------------------------------------------------------------------------------------------------------------------------------------
```

File: docs/source/release/0.61.0-notes.rst [hunk 28]
```diff
@@ -0,0 +1,457 @@
...
+Use of dead branch pruning improved
+-----------------------------------
+
+Dead-branch pruning use is improved to support cases when the predicate
+expression is dependent on a variable that later changes type.
+
+(`PR-#9711 <https://github.com/numba/numba/pull/9711>`__)
+
+``find_topo_sort`` reliabililty improvement
+------------------------------------------
```
Comment (esc) [inline docs/source/release/0.61.0-notes.rst:165]:
```suggestion
-------------------------------------------
```

**Review by esc — APPROVED** [review]
Second pass, clearing up all the things I missed during my first pass.

**Review by esc — CHANGES_REQUESTED** [review]
Sorry, wrong button during last review, still needs changes.

File: docs/source/release/0.61.0-notes.rst (hunk 28 above)
Comment (esc) [reply docs/source/release/0.61.0-notes.rst:165]: This will fix the sphinx build error too.

**Review by esc — APPROVED** [review]
LGTM, I did another quick review and it looks fine.

**Pull request merged**